derive/src/utils.rs

```diff
@@ -179,22 +189,40 @@ pub fn codec_crate_path(attrs: &[Attribute]) -> syn::Result<Path> {
 	}
 }
 
-/// Trait bounds.
-pub type TraitBounds = Punctuated<syn::WherePredicate, token::Comma>;
-
-/// Parse `name(T: Bound, N: Bound)` as a custom trait bound.
-struct CustomTraitBound<N> {
-	_name: N,
-	_paren_token: token::Paren,
-	bounds: TraitBounds,
+/// Parse `name(T: Bound, N: Bound)` or `name(skip_type_params(T, N))` as a custom trait bound.
+pub enum CustomTraitBound<N> {
+	SpecifiedBounds {
+		_name: N,
+		_paren_token: token::Paren,
+		bounds: Punctuated<syn::WherePredicate, token::Comma>,
+	},
+	SkipTypeParams {
+		_name: N,
+		_paren_token_1: token::Paren,
+		_skip_type_params: skip_type_params,
+		_paren_token_2: token::Paren,
+		type_params: Punctuated<TypeParam, Token![,]>,
```

Comment:
Parsing it as a `TypeParam` is actually much easier, as I can just access the `ident` field on `TypeParam`, rather than going through layers of matching just to get to the type identifier.

Reply:
Good point.

Reply:
But then the user could also just provide wrong data. Could be not actually only parse `Ident` here?

Reply:
Yeah, I guess that makes sense as well.